### PR TITLE
Add basic access control to /reaper/

### PR DESCRIPTION
**Pull request by kkellerlbl**
Compare remote_addr to reaper_allowed_ip, forbid request if it doesn't match.  (Can use 0.0.0.0 to allow /reaper/ from anywhere.)

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -601,14 +602,22 @@ def reaper_endpoint():
...
-        resp.status_code=200
-    except Exception as ex:
-        resp = flask.Response("Reaper error: {}".format(repr(ex)))
-        resp.status_code=500
+
+    request = flask.request
+    logger.info({"message": "Reaper endpoint called from {}".format(request.remote_addr)})
+
+    resp = flask.Response("generic Reaper response")
+    if (cfg['reaper_allowed_ip'] == request.remote_addr or cfg['reaper_allowed_ip'] == "0.0.0.0"):
```
Comment (sychan) [inline app.py]: Maybe change it to a prefix match using startswith()? That way we could restrict it to a subnet if we wanted to?

**Review by sychan — COMMENTED** [review]
How about making it so that a prefix match works?

File: app.py (hunk 1 above)
Comment (kkellerlbl) [reply app.py]: If we want to do that we should just do real CIDR matching instead.

Comment (kkellerlbl) [reply app.py]: Added ipaddress module to support one subnet.

File: app.py [hunk 2]
```diff
@@ -601,14 +603,22 @@ def reaper_endpoint():
...
-        resp = flask.Response("Reaper success: {} deleted".format(num))
-        resp.status_code=200
-    except Exception as ex:
-        resp = flask.Response("Reaper error: {}".format(repr(ex)))
-        resp.status_code=500
+
+    request = flask.request
+    logger.info({"message": "Reaper endpoint called from {}".format(request.remote_addr)})
+
+    resp = flask.Response("generic Reaper response")
```
Comment (sychan) [inline app.py]: Want to delete this? It is redundant.

**Review by sychan — CHANGES_REQUESTED** [review]

**Pull request merged**